libp2p/switch.nim

```diff
@@ -237,18 +237,24 @@ proc dial*(s: Switch,
               asyncCheck s.cleanupConn(conn)
           break
   else:
-    trace "Reusing existing connection"
+    trace "Reusing existing connection", id = conn.id
 
   if isNil(conn):
     raise newException(CatchableError, "Unable to establish outgoing link")
+  else:
+    result = conn
 
   if proto.len > 0 and not conn.closed:
+    # TODO: avoid naming confusion. Muxer streams are somehow Connection
```

Comment:
`LPChannel` inherits from `LPStream`, the point of having one `Connection` type as opposed to distinguishing between different types of streams in upper layers of the library is to allow for many types of streams to be consumed without changes. This allows using multiplexed stream or pure connections without changing anything else in the connection handling code.

Reply:
But `getMuxedStream()` does not return a stream of any kind. It returns a `Future[Connection]`. So what does "stream" mean here?

Reply:
A `Connection` is also a stream since it's derived from `LPStream`, however,  it's called stream also because I explicitly request it from the muxer.